engine/src/main/java/com/agiletec/aps/system/common/AbstractSearcherDAO.java

```diff
@@ -32,464 +32,469 @@
+                    parameter = "%" + ((String) object) + "%";
+                } else if (likeOptionType.equals(FieldSearchFilter.LikeOptionType.LEFT)) {
+                    parameter = "%" + ((String) object);
+                } else if (likeOptionType.equals(FieldSearchFilter.LikeOptionType.RIGHT)) {
+                    parameter = ((String) object) + "%";
+                }
+                stat.setString(index, parameter);
+            } else {
+                stat.setString(index, (String) object);
+            }
+        } else if (object instanceof Date) {
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime((Date) object);
+            if (dateDelay != null) {
+                calendar.add(Calendar.DATE, dateDelay);
+            }
+            if (object instanceof Timestamp) {
+                Timestamp timestamp = new Timestamp(calendar.getTime().getTime());
+                stat.setTimestamp(index, timestamp);
+            } else {
+                Date data = calendar.getTime();
+                stat.setDate(index, new java.sql.Date(data.getTime()));
+            }
+        } else if (object instanceof BigDecimal) {
+            stat.setBigDecimal(index, (BigDecimal) object);
+        } else if (object instanceof Boolean) {
+            stat.setString(index, ((Boolean) object).toString());
+        } else {
+            stat.setObject(index, object);
+        }
+    }
+
+    protected String createQueryString(FieldSearchFilter[] filters, boolean isCount, boolean selectAll) {
+        StringBuffer query = this.createBaseQueryBlock(filters, isCount, selectAll);
+
+        boolean hasAppendWhereClause = false;
+        if (isCount) {
+            hasAppendWhereClause = this.appendMetadataFieldFilterQueryBlocksForCount(filters, query, false);
+        }
+
+        if (!isCount) {
+            hasAppendWhereClause = this.appendMetadataFieldFilterQueryBlocks(filters, query, false);
+            this.appendLimitQueryBlock(filters, query, hasAppendWhereClause);
+            boolean ordered = appendOrderQueryBlocks(filters, query, false);
+        }
+
+        return query.toString();
+    }
+
+    /**
+     * Create the 'base block' of the query with the eventual references to the support table.
+     * @param filters The filters defined.
+     * @param selectAll When true, this will insert all the fields in the master table in the select 
+     * of the master query.
+     * When true we select all the available fields; when false only the field addressed by the filter
+     * is selected.
+     * @return The base block of the query.
+     */
+    protected StringBuffer createBaseQueryBlock(FieldSearchFilter[] filters, boolean selectAll) {
+        return this.createBaseQueryBlock(filters, false, selectAll);
+    }
+
+    protected StringBuffer createBaseQueryBlock(FieldSearchFilter[] filters, boolean isCount, boolean selectAll) {
+        StringBuffer query = null;
+        if (isCount) {

```

Comment:
If statement not needed. Calls are the same.

If the calls are the same is the method needed.